<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SignalDetailsDialog</class>
 <widget class="QDialog" name="SignalDetailsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Signal Details</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Dark theme stylesheet */
QDialog, QWidget {
    background-color: #1e1e1e;
    color: #f0f0f0;
}

QLabel {
    color: #f0f0f0;
}

QFrame {
    background-color: #2d2d2d;
    border-radius: 5px;
}

QPushButton {
    background-color: #0d6efd;
    color: white;
    border: none;
    padding: 6px 12px;
    border-radius: 4px;
}

QPushButton:hover {
    background-color: #0b5ed7;
}

QPushButton:pressed {
    background-color: #0a58ca;
}

QPushButton:checked {
    background-color: #0a58ca;
}

/* Navigation button styling */
#sidebar_widget QPushButton {
    background-color: #2d2d2d;
    color: #f0f0f0;
    border-radius: 4px;
    text-align: left;
    padding-left: 15px;
    border: none;
}

#sidebar_widget QPushButton:hover {
    background-color: #3d3d3d;
}

#sidebar_widget QPushButton:checked {
    background-color: #f0f0f0;
    color: #1e1e1e;
    font-weight: bold;
}

#sidebar_widget {
    background-color: #1e1e1e;
    border-right: 1px solid #333333;
}

QStackedWidget {
    background-color: #1e1e1e;
}

QTreeWidget, QTableWidget {
    background-color: #2d2d2d;
    alternate-background-color: #363636;
    border: 1px solid #3d3d3d;
    gridline-color: #3d3d3d;
}

QHeaderView::section {
    background-color: #3d3d3d;
    color: #f0f0f0;
    padding: 5px;
    border: 1px solid #505050;
}

QLineEdit, QPlainTextEdit, QSpinBox, QDoubleSpinBox, QComboBox, QDateTimeEdit {
    background-color: #363636;
    border: 1px solid #505050;
    color: #f0f0f0;
    padding: 4px;
    border-radius: 3px;
}

QLineEdit:focus, QPlainTextEdit:focus, QSpinBox:focus, QDoubleSpinBox:focus, QComboBox:focus, QDateTimeEdit:focus {
    border: 1px solid #0d6efd;
}

QScrollBar:vertical {
    background-color: #2d2d2d;
    width: 12px;
    margin: 0px;
}

QScrollBar::handle:vertical {
    background-color: #505050;
    min-height: 20px;
    border-radius: 6px;
}

QScrollBar::handle:vertical:hover {
    background-color: #0d6efd;
}

QScrollBar:horizontal {
    background-color: #2d2d2d;
    height: 12px;
    margin: 0px;
}

QScrollBar::handle:horizontal {
    background-color: #505050;
    min-width: 20px;
    border-radius: 6px;
}

QScrollBar::handle:horizontal:hover {
    background-color: #0d6efd;
}

QScrollBar::add-line, QScrollBar::sub-line {
    border: none;
    background: none;
}

QScrollBar::add-page, QScrollBar::sub-page {
    background: none;
}

QGroupBox {
    border: 1px solid #505050;
    border-radius: 5px;
    margin-top: 1ex;
    font-weight: bold;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top center;
    padding: 0 5px;
    color: #f0f0f0;
}

QDialogButtonBox {
    background-color: #1e1e1e;
    border-top: 1px solid #505050;
    padding-top: 15px;
}</string>
  </property>
  <layout class="QHBoxLayout" name="main_layout">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QWidget" name="sidebar_widget" native="true">
     <property name="minimumSize">
      <size>
       <width>180</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>180</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="styleSheet">
      <string notr="true">QWidget#sidebar_widget {
    background-color: #1e1e1e;
    color: #f0f0f0;
}</string>
     </property>
     <layout class="QVBoxLayout" name="sidebar_layout">
      <property name="spacing">
       <number>5</number>
      </property>
      <property name="leftMargin">
       <number>10</number>
      </property>
      <property name="topMargin">
       <number>20</number>
      </property>
      <property name="rightMargin">
       <number>10</number>
      </property>
      <property name="bottomMargin">
       <number>20</number>
      </property>
      <item>
       <widget class="QLabel" name="signal_name_label">
        <property name="font">
         <font>
          <pointsize>10</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string notr="true">font-size: 14px;
font-weight: bold;
color: #f0f0f0;
margin-bottom: 20px;</string>
        </property>
        <property name="text">
         <string>SignalName</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout" name="nav_buttons_layout">
        <property name="spacing">
         <number>5</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QPushButton" name="navButton0">
          <property name="text">
           <string>Basic Properties</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="navButton1">
          <property name="text">
           <string>Advanced Properties</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="navButton2">
          <property name="text">
           <string>Core Routing</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="content_stack">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="page_basic">
      <layout class="QVBoxLayout" name="basic_tab_layout">
       <property name="leftMargin">
        <number>20</number>
       </property>
       <property name="topMargin">
        <number>20</number>
       </property>
       <property name="rightMargin">
        <number>20</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item>
        <layout class="QFormLayout" name="basic_form_layout">
         <property name="horizontalSpacing">
          <number>15</number>
         </property>
         <property name="verticalSpacing">
          <number>15</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="signal_name_label_text">
           <property name="text">
            <string>Signal Name:</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLabel" name="signal_name_display">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>SignalName</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="var_port_name_label">
           <property name="text">
            <string>Variable Port Name:</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="variable_port_name_edit"/>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="data_type_label">
           <property name="text">
            <string>Data Type:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QComboBox" name="data_type_combo">
           <property name="editable">
            <bool>true</bool>
           </property>
           <item>
            <property name="text">
             <string>INT8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>UINT8</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>INT16</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>UINT16</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>INT32</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>UINT32</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>INT64</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>UINT64</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>FLOAT32</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>FLOAT64</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>BOOLEAN</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>CHAR</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>STRING</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>STRUCT</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="memory_region_label">
           <property name="text">
            <string>Memory Region:</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QComboBox" name="memory_region_combo">
           <item>
            <property name="text">
             <string>DDR</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Cached</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>NonCached</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="type_label">
           <property name="text">
            <string>Type:</string>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <widget class="QComboBox" name="type_combo">
           <item>
            <property name="text">
             <string>Concurrent</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Sequential</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="init_value_label">
           <property name="text">
            <string>Init Value:</string>
           </property>
          </widget>
         </item>
         <item row="8" column="1">
          <widget class="QComboBox" name="init_value_combo">
           <item>
            <property name="text">
             <string>ZeroMemory</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Custom</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string/>
           </property>
          </widget>
         </item>
         <item row="9" column="1">
          <widget class="QPushButton" name="custom_value_button">
           <property name="visible">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Enter Custom Value...</string>
           </property>
          </widget>
         </item>
         <item row="10" column="0">
          <widget class="QLabel" name="asil_label">
           <property name="text">
            <string>ASIL:</string>
           </property>
          </widget>
         </item>
         <item row="10" column="1">
          <widget class="QComboBox" name="asil_combo">
           <item>
            <property name="text">
             <string>QM</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>A</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>B</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>C</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>D</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QGroupBox" name="struct_group">
         <property name="visible">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Structure Definition</string>
         </property>
         <layout class="QVBoxLayout" name="struct_layout">
          <item>
           <widget class="QLabel" name="struct_fields_label">
            <property name="text">
             <string>Structure Fields:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QTreeWidget" name="struct_fields_tree">
            <property name="alternatingRowColors">
             <bool>true</bool>
            </property>
            <column>
             <property name="text">
              <string>Field Name</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Data Type</string>
             </property>
            </column>
            <column>
             <property name="text">
              <string>Description</string>
             </property>
            </column>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="field_buttons_layout">
            <item>
             <widget class="QPushButton" name="add_field_button">
              <property name="text">
               <string>Add Field</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="edit_field_button">
              <property name="text">
               <string>Edit Field</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="remove_field_button">
              <property name="text">
               <string>Remove Field</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_advanced">
      <layout class="QVBoxLayout" name="advanced_tab_layout">
       <property name="leftMargin">
        <number>20</number>
       </property>
       <property name="topMargin">
        <number>20</number>
       </property>
       <property name="rightMargin">
        <number>20</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item>
        <widget class="QFrame" name="card">
         <layout class="QVBoxLayout" name="advanced_card_layout">
          <item>
           <layout class="QFormLayout" name="advanced_form_layout">
            <property name="horizontalSpacing">
             <number>15</number>
            </property>
            <property name="verticalSpacing">
             <number>15</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="buffer_count_ipc_label">
              <property name="text">
               <string>Buffer Count IPC:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="buffer_count_ipc_spin">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="impl_approach_label">
              <property name="text">
               <string>Implementation Approach:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="impl_approach_combo">
              <item>
               <property name="text">
                <string>SharedMemory</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>VRING</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>IpcOvEth</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="get_obj_ref_label">
              <property name="text">
               <string>Get Object Reference:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="get_obj_ref_combo">
              <item>
               <property name="text">
                <string>False</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>True</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="notifiers_label">
              <property name="text">
               <string>Notifiers:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QComboBox" name="notifiers_combo">
              <item>
               <property name="text">
                <string>False</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>True</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="sm_buff_count_label">
              <property name="text">
               <string>SM Buffer Count:</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QSpinBox" name="sm_buff_count_spin">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="timeout_label">
              <property name="text">
               <string>Timeout:</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QSpinBox" name="timeout_spin">
              <property name="suffix">
               <string> ms</string>
              </property>
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>200</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="periodicity_label">
              <property name="text">
               <string>Periodicity:</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QSpinBox" name="periodicity_spin">
              <property name="suffix">
               <string> ms</string>
              </property>
              <property name="minimum">
               <number>10</number>
              </property>
              <property name="maximum">
               <number>200</number>
              </property>
              <property name="singleStep">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QLabel" name="checksum_label">
              <property name="text">
               <string>Checksum:</string>
              </property>
             </widget>
            </item>
            <item row="7" column="1">
             <widget class="QComboBox" name="checksum_combo">
              <item>
               <property name="text">
                <string>None</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Additive</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>CustomChecksum</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="page_routing">
      <layout class="QVBoxLayout" name="routing_tab_layout">
       <property name="leftMargin">
        <number>20</number>
       </property>
       <property name="topMargin">
        <number>20</number>
       </property>
       <property name="rightMargin">
        <number>20</number>
       </property>
       <property name="bottomMargin">
        <number>20</number>
       </property>
       <item>
        <widget class="QScrollArea" name="core_scroll_area">
         <property name="widgetResizable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="scroll_content">
          <property name="geometry">
           <rect>
            <x>0</x>
            <y>0</y>
            <width>556</width>
            <height>505</height>
           </rect>
          </property>
          <layout class="QVBoxLayout" name="scroll_layout">
           <item>
            <widget class="QGroupBox" name="source_group">
             <property name="title">
              <string>Source Core</string>
             </property>
             <layout class="QVBoxLayout" name="source_layout">
              <item>
               <widget class="QComboBox" name="source_combo">
                <item>
                 <property name="text">
                  <string>&lt;None&gt;</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="dest_group">
             <property name="title">
              <string>Destination Cores</string>
             </property>
             <layout class="QVBoxLayout" name="dest_layout"/>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
  <widget class="QWidget" name="button_widget" native="true">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>560</y>
     <width>800</width>
     <height>40</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="button_layout">
    <item>
     <spacer name="horizontalSpacer">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>40</width>
        <height>20</height>
       </size>
      </property>
     </spacer>
    </item>
    <item>
     <widget class="QPushButton" name="ok_button">
      <property name="text">
       <string>OK</string>
      </property>
      <property name="default">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QPushButton" name="cancel_button">
      <property name="text">
       <string>Cancel</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
